src/redux/services/helm.ts

```diff
@@ -79,20 +82,81 @@ export function isHelmChartFolder(files: string[]): boolean {
  * Adds the values file at the given path to the specified HelmChart
  */
 
-export function createHelmValuesFile(fileEntry: FileEntry, helmChart: HelmChart, helmValuesMap: HelmValuesMapType) {
+interface CreateHelmValuesFileParams {
+  fileEntry: FileEntry;
+  helmChart: HelmChart;
+  helmValuesMap: HelmValuesMapType;
+  fileMap: FileMapType;
+}
+
+const get = (t: object, objPath: string) => objPath.split('.').reduce((r, k) => (r as any)?.[k], t);
+
+const getYamlScalar = (contents: any, keyPath: string): Scalar | undefined => {
+  const keyParts = keyPath.split('.');
+  const keyStart = keyParts.shift();
+  const pair = contents.items.find((item: any) => {
+    return item.key.value === keyStart;
+  });
+  if (!pair) {
+    return;
+  }
+
+  if (!keyParts.length) {
+    return pair.value;
+  }
+
+  return getYamlScalar(pair.value, keyParts.join('.'));
+};
+
+export function createHelmValuesFile({fileEntry, helmChart, helmValuesMap, fileMap}: CreateHelmValuesFileParams) {
+  const filePath = getAbsoluteFilePath(fileEntry.filePath, fileMap);
+  const fileContent = fs.readFileSync(filePath, 'utf8');
+  const lineCounter = new LineCounter();
+  const documents = parseAllYamlDocuments(fileContent, lineCounter);
+
+  const values: HelmValueMatch[] = [];
+  documents.forEach((doc: any) => {
+    const helmObject = doc.toJS();
+    getObjectKeys(helmObject).forEach(keyPath => {
+      const scalar = getYamlScalar(doc.contents, keyPath);
```

Comment:
I think you use the build-in `doc.getIn(KeyPath, true)` instead.

Reply:
Tried that but it does not work as expected :(